packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -196,29 +190,297 @@ class CupertinoAlertDialog extends StatelessWidget {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      MultiChildLayoutParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    final double contentWidth = _content.getMaxIntrinsicWidth(height);
+    final double actionsWidth = _actions.getMaxIntrinsicWidth(height);
+    final double width = max(contentWidth, actionsWidth);
+
+    if (width.isFinite)
+      return width;
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      CupertinoDialogRenderBoxParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+
+    final double contentWidth = _content.getMaxIntrinsicWidth(height);
+    final double actionsWidth = _actions.getMaxIntrinsicWidth(height);
+    final double width = min(contentWidth, actionsWidth);
+
+    if (width.isFinite)
+      return width;
+    return 0.0;
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      CupertinoDialogRenderBoxParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    final double contentHeight = _content.getMaxIntrinsicHeight(width);
+    final double actionsHeight = _actions.getMaxIntrinsicHeight(width);
+    final double height = max(contentHeight, actionsHeight);
+
+    if (height.isFinite)
+      return height;
+    return 0.0;
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    RenderBox _content;
+    RenderBox _actions;
+    getChildrenAsList().forEach((RenderBox child) {
+      CupertinoDialogRenderBoxParentData parentData = child.parentData;
+      if (parentData.id == _AlertDialogSection.content) {
+        _content = child;
+      } else if (parentData.id == _AlertDialogSection.actions) {
+        _actions = child;
+      }
+    });
+    assert(_content != null);
+    assert(_actions != null);
+
+    final double contentHeight = _content.getMaxIntrinsicHeight(width);
+    final double actionsHeight = _actions.getMaxIntrinsicHeight(width);
+    final double height = min(contentHeight, actionsHeight);
```

Comment:
Actually is this right? For its max intrinsic height, you're reporting the smaller of the content or the actions's height? 

Reading the doc for this method, since your width doesn't depend on your height, I think your max and min intrinsic heights should be the same.

Reply:
I guess that should actually be contentHeight+actionsHeight, right?

This is another reason why a render object seems like a strange place for this.  We compute all relevant sizing in the layout method, but we also have to implement these other methods....and it seems they're so unrelated that even a bad implementation doesn't effect the dialog....